src/config_consumer.rs

```diff
@@ -0,0 +1,232 @@
+            ?err,
+            "Key must be a valid subscription_id UUID. Got {:?}", key
+        );
+        InvalidMessage::from(broker_message)
+    })?;
+
+    // TODO(epurkhiser): We may want to use a batching strategy here to avoid lock contention on
+    // the ConfigStore.
+
+    match broker_message.payload.payload() {
+        // Register new configuration
+        Some(payload) => {
+            let mut config: CheckConfig = rmp_serde::from_slice(payload).map_err(|err| {
+                error!(?err, "Failed to decode config message");
+                InvalidMessage::from(broker_message)
+            })?;
+
+            if config.subscription_id != subscription_id {
+                error!(?key, ?config.subscription_id, "Config key mismatch");
+                return Err(InvalidMessage::from(broker_message));
+            }
+
+            config.partition = broker_message.partition.index;
+
+            // Store configuration
+            debug!(config = ?config, "Consumed configuration");
+            config_store
+                .write()
+                .expect("Lock poisoned")
+                .add_config(Arc::new(config));
+        }
+        // Remove existing configuration
+        None => config_store
+            .write()
+            .expect("Lock poisoned")
+            .remove_config(subscription_id),
+    }
+
+    Ok(message)
+}
+
+struct ConfigConsumerFactory {
+    config_store: Arc<RwConfigStore>,
+}
+
+impl ProcessingStrategyFactory<KafkaPayload> for ConfigConsumerFactory {
+    fn create(&self) -> Box<dyn ProcessingStrategy<KafkaPayload>> {
+        let config_store = self.config_store.clone();
+
+        Box::new(RunTask::new(
+            move |message| update_config_store(config_store.clone(), message),
+            Noop {},
+        ))
+    }
+
+    fn update_partitions(&self, partitions: &HashMap<Partition, u64>) {
+        // Revoke configurations for partitions that are no longer assigned to this consumer.
+        self.config_store
+            .write()
+            .expect("Lock poisoned")
+            .update_partitions(&partitions.keys().map(|p| p.index).collect())
+    }
+}
+
+/// Runs a kafka consumer to read configurations from the uptime-configs topic. Each new config is
+/// added to the ['ConfigStore'] as it is recieved. Null mesages will revoke configs.
+pub fn run_config_consumer(
+    topic_name: String,
+    consumer_config: KafkaConfig,
+    config_store: Arc<RwConfigStore>,
+    shutdown: CancellationToken,
+) -> JoinHandle<()> {
+    let stream_processor = StreamProcessor::with_kafka(
+        consumer_config,
+        ConfigConsumerFactory { config_store },
+        Topic::new(&topic_name),
+        None,
+    );
+
+    let mut processing_handle = stream_processor.get_handle();
+
+    let join_handle = thread::spawn(move || {
+        info!("Starting config consumer");
+        stream_processor
+            .run()
+            .expect("Failed to run config consumer");
+    });
```

Comment:
Spawn the consumer in it's own thread outside of tokio, no need to treat it as a future.